### ISSUE TITLE
Crash on SAF tag

### ISSUE THREAD

**Issue by shuvashish76** [MuntashirAkon/AppManager#1294]
### Please check before submitting an issue

- [X] I know what my device, OS and App Manager versions are
- [X] I know how to take logs
- [X] I know how to reproduce the issue which may not be specific to my device

### Describe the bug

Open App Info page of ~any app~ e.g. Aard 2, KeePassDX, click on SAF tag to view list of granted URIs. App crashes.

Edit: Checked with few other apps e.g. KDE Connect, Material Files, Tasks.org, it doesn't crash for these apps.

### Logs

```
java.lang.IndexOutOfBoundsException
    at android.net.Uri$PathSegments.get(Uri.java:1012)
    at android.net.Uri$PathSegments.get(Uri.java:997)
    at io.github.muntashirakon.AppManager.uri.GrantUriUtils.toLocalisedString(GrantUriUtils.java:32)
    at io.github.muntashirakon.AppManager.details.info.AppInfoFragment.lambda$getTagCloudItems$61$io-github-muntashirakon-AppManager-details-info-AppInfoFragment(AppInfoFragment.java:991)
    at io.github.muntashirakon.AppManager.details.info.AppInfoFragment$$ExternalSyntheticLambda75.onClick(Unknown Source:4)
    at android.view.View.performClick(View.java:6597)
    at android.widget.CompoundButton.performClick(CompoundButton.java:134)
    at android.view.View.performClickInternal(View.java:6574)
    at android.view.View.access$3100(View.java:778)
    at android.view.View$PerformClick.run(View.java:25906)
    at android.os.Handler.handleCallback(Handler.java:873)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loop(Looper.java:193)
    at android.app.ActivityThread.main(ActivityThread.java:6718)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

### Device info

App version: **4.0.0-alpha02-DEBUG** (AM Debug v4.0.0-alpha02 Run#2644)
App version code: 435
Android build version: eng.chef.20200713.073313
Android release version: 9
Android SDK version: 28
Android build ID: lineage_TB8704-userdebug 9 PQ3A.190801.002 eng.chef.20200713.073313 test-keys
Device brand: Lenovo
Device manufacturer: Lenovo
Device name: TB-8704X
Device model: Lenovo TB-8704X
Device product name: lineage_TB8704
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-IN
In-App Language: auto
Mode: root
Inferred Mode: Root

### Additional context

No crash for _any_ SAF tagged apps for AM v3.1.4 _stable_ release.

**Comment by MuntashirAkon** [comment]
Is the issue still present?

**Comment by shuvashish76** [comment]
AM Debug v4.0.0-alpha02 Run#2654
Yes

**Comment by MuntashirAkon** [comment]
Try run#2659. You should get a different crash.

**Comment by shuvashish76** [comment]
For Aard 2
```
FATAL EXCEPTION: main
Process: io.github.muntashirakon.AppManager.debug, PID: 5670
java.lang.UnsupportedOperationException: Unsupported URI: content://com.android.externalstorage.documents/document/primary%3ASyncthing%2FDictionaries%2Fwordnet-3.1.slob
	at io.github.muntashirakon.AppManager.uri.GrantUriUtils.toLocalisedString(GrantUriUtils.java:35)
	at io.github.muntashirakon.AppManager.details.info.AppInfoFragment.lambda$getTagCloudItems$61$io-github-muntashirakon-AppManager-details-info-AppInfoFragment(AppInfoFragment.java:991)
	at io.github.muntashirakon.AppManager.details.info.AppInfoFragment$$ExternalSyntheticLambda75.onClick(Unknown Source:4)
	at android.view.View.performClick(View.java:6597)
	at android.widget.CompoundButton.performClick(CompoundButton.java:134)
	at android.view.View.performClickInternal(View.java:6574)
	at android.view.View.access$3100(View.java:778)
	at android.view.View$PerformClick.run(View.java:25906)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6718)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```
For KeePassDX
<details><summary>Details</summary>
<p>

```
FATAL EXCEPTION: main
Process: io.github.muntashirakon.AppManager.debug, PID: 8989
java.lang.UnsupportedOperationException: Unsupported URI: content://me.zhanghai.android.files.file_provider/file%253A%252F%252F%252Fstorage%252Femulated%252F0%252FPictures%252Fkeyfile2.keyx
	at io.github.muntashirakon.AppManager.uri.GrantUriUtils.toLocalisedString(GrantUriUtils.java:35)
	at io.github.muntashirakon.AppManager.details.info.AppInfoFragment.lambda$getTagCloudItems$61$io-github-muntashirakon-AppManager-details-info-AppInfoFragment(AppInfoFragment.java:991)
	at io.github.muntashirakon.AppManager.details.info.AppInfoFragment$$ExternalSyntheticLambda75.onClick(Unknown Source:4)
	at android.view.View.performClick(View.java:6597)
	at android.widget.CompoundButton.performClick(CompoundButton.java:134)
	at android.view.View.performClickInternal(View.java:6574)
	at android.view.View.access$3100(View.java:778)
	at android.view.View$PerformClick.run(View.java:25906)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6718)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

</p>
</details>

**Comment by MuntashirAkon** [comment]
Fixed in 316b5f47315422cf33f4427b6a536798a768b1ec
